### ISSUE TITLE
Feature request: tools to display and set zoom factor for ratio screen pixel / underlying data pixel

### ISSUE THREAD

**Issue by VolkerH** [napari/napari#2145]
## 🚀 Feature

This has briefly been discussed in the Zulip chat. 

Currently, the napari interface does not give any indication whether the current view is zoomed in or zoomed out and by how far.
In particular it would be useful to know that one is looking at the data at 100% zoom, that is one screen pixel corresponds directly to one data pixel (provided there are no affines that cause interpolation). 

With zoom factor here I refer to the ratio screen pixels per data pixel and not to some scaled representation (screen millimeter to calibrated pixel millimeter) as it may be present in some graphic design software.

Additionaly, it would also be good to have shorthands (either in the menu or keyboard shortcuts or both) to go to 100% and maybe to increase the zooom level in fixed step sizes.

For illustration, here are some screenshots from a different viewer:

![image](https://user-images.githubusercontent.com/2210044/105466602-c1b96500-5c94-11eb-83ce-02389bf90132.png)

![image](https://user-images.githubusercontent.com/2210044/105466731-f75e4e00-5c94-11eb-9900-0d9ebc29dd6a.png)

I actually don't know what the difference between `View 100%` and `Original Scale` is in that other viewer.

## Motivation

One example where this feature would be useful is inspecting individiual greyvalues with the mouse cursor. One would like to know that one is at 100% or maybe at 200% and not at a value such as 33% 

## Additional context

As @sofroniewn suggested on zulip, the relevant information for a given layer can be obtained as `viewer.camera.zoom * layer.scale`.
I think the place where this should be displayed is next to the cursor location and intensity value. 
Any preferences whether 1:1 should be indicated as `100%` or `1.0` ?

Also note that other features like `scale to fit` may be useful, although in the context of napari I would call these `set zoom to fit` in  a departure from many other software tools (just to make clear the underlying data represantation is not scaled).

**Comment by sofroniewn** [comment]
This request sounds very reasonable - I think we just need to make sure we have space in the status bar, or maybe we now think of making an additional dock widget that has lots of status information that people can show and hide. We could even start that as a simple `napari-status` plugin and decide what we like/ don't like / want to promote to main status bar. Doesn't have to be done that way, but is just a thought!

**Comment by VolkerH** [comment]
Ok, I started looking and found that the status message is set in the `get_message` method of the `Layer` class:
https://github.com/napari/napari/blob/3631b159f84613216e8516c833b951d89dc2b5da/napari/layers/base/base.py#L979-L991
However, I am not sure how I can access `camera.zoom` from this method. Does a layer class hold some sort of reference to its `viewer` ?

**Comment by sofroniewn** [comment]
> Ok, I started looking and found that the status message is set in the get_message method of the Layer class:
> However, I am not sure how I can access camera.zoom from this method. Does a layer class hold some sort of reference to its viewer ?

Yeah, it's a little tricky right now. I've got PR #1985 that would deprecate layer.status and make all of this more unified. It will be much easier to do the update then. I can go through that PR, fix any merge conflicts and then maybe @jni can give it some review!

**Comment by VolkerH** [comment]
Hi @sofroniewn 

I just browsed through the changes introduced in #1985. While I see changes to the way the status is updated for a layer and the layer generates its message, I still don't see a way how, given a layer, one can access its viewer and hence the camera, which would be required to show the current zoom. Unless I missed it, the layer doesn't hold a reference to its viewer. I assume that might be (somewhat) intentional, as in  keeping the data model separate from the viewer.
I don't really know enough about the architecture to see how to resolve this in a clean way. 

When thinking about this further, if there ever is multi-canvas support, there could be several viewers referencing the same data with different zoom levels. So in this case it should really be the viewer that generates the status messages and not the layer that holds the data.

**Comment by sofroniewn** [comment]
> So in this case it should really be the viewer that generates the status messages and not the layer that holds the data.

Yes exactly, now I think I would do something here inside [_on_cursor_position_change](https://github.com/napari/napari/blob/v0.4.4/napari/components/viewer_model.py#L409-L411) in viewer_model.py like

```python
            layer_status = self.active_layer.get_status(
                self.cursor.position, world=True
            )
            zoom_status = f'Camera zoom {self.camera.zoom}`
            # or something like 
            # zoom_status = f'Total zoom {self.camera.zoom * self.active_layer.scale}`
           self.status = zoom_status + layer_status
```

My only two fears are

(1) `active_layer.scale` might now get called everyntime the mouse moves, which is not so good as it is calculated via the affine decomposition now, so we should check how long that takes, if it if `us` then that's fine, but if it is `ms` then that isn't good.

(2) Is this of sufficient general interested that we should now display it to all users all the time? I worry the status bar messages might be getting longer and longer (we added support for labels properties recently for example).

One longer term option is people could set what they wanted in the status bar in the settings and preferences. Maybe we'd have a list of 5 things they could select and then depending on the configuration they had selected we'd show them the different combination.

If we like that idea we could temporarily add a private `_show_zoom_status` bool to the `ViewerModel` and then depending on if it is True or False we could append this status information or not.

Curious if @jni has other ideas?
